lib/internal/errors.js

```diff
@@ -1471,8 +1476,20 @@ E('ERR_NETWORK_IMPORT_DISALLOWED',
   "import of '%s' by %s is not supported: %s", Error);
 E('ERR_NOT_BUILDING_SNAPSHOT',
   'Operation cannot be invoked when not building startup snapshot', Error);
-E('ERR_NOT_SUPPORTED_IN_SNAPSHOT',
-  '%s is not supported in startup snapshot', Error);
+E('ERR_NOT_SUPPORTED_IN_SNAPSHOT', function(reason) {
+  hideInternalStackFrames(this);
+  const msg = `${reason} is not supported in startup snapshot`;
+  ObjectDefineProperties(this, {
```

Comment:
I don't think special casing this so that one test can pass is enough. This can happen to more than just this error and in code paths other than snapshotting (e.g. whenever `env->can_call_into_js()` is false in `TriggerUncaughtException` or `ReportFatalException()`. This should be done for all the errors to make sure that whenever it falls to the C++ land to format the error and the C++ land cannot call into JS, the message is still available as a data property.